scipy/stats/tests/test_morestats.py

```diff
@@ -1163,6 +1163,18 @@ def test_all(self):
         maxlog_all = stats.boxcox_normmax(self.x, method='all')
         assert_allclose(maxlog_all, [1.804465, 1.758101], rtol=1e-6)
 
+    def test_single_value(self):
```

Comment:
you need to check actual outcome against expected outcome using one of the `assert_...` functions

Reply:
The actual outcome is irrelevant and to be considered undefined. These tests simply check that no warnings / errors are thrown. Without my changes, these tests fail.

Reply:
The actual outcome is irrelevant and to be considered undefined. These tests simply check that no warnings / errors are thrown. Without my changes, these tests fail.

Reply:
The output has to be checked regardless. Running the code successfully is already checking for exceptions.

Reply:
What should the output be? Right now, it is dependent on implementation details of the underlying functions all the way to the optimizer. I'm afraid that checking the output will be testing the implementation and not the specification.

The only thing I had in mind for these tests was to check for exceptions.

Reply:
> You can also call these private functions in the tests. 

Unfortunately, some of the functions are defined dynamically (def within def). Also, how would the higher level function be tested if I replace this test with several tests checking the private ones? Right now, there's a division by zero happening with the inputs chosen in the tests I added. Personally I'd prefer to make sure the test case passes with the higher level function rather than testing the private functions which are subject to change.

The tricky bit is that, in my opinion, the only reasonable thing to check for these test cases is that there's no error. The output might change if other private (shared) functions are changed, but that does not make the results of the higher level function less valid, and I'm trying not to test the implementation details.

Reply:
All test cases you added are degenerate in the sense that there cannot be sensible value for lambda If I am not mistaken (single value input or all ones). In #6873, there were more "realistic" inputs and one could also test the outcome against R in that case. I think auch tests are better suited to check if the issue has been fixed.

Reply:
As requested, I added `test_not_equal_output_values` using the input from one of the comments in the original isssue.

Reply:
As requested, I added `test_not_equal_output_values` with values from the original issue.